ftprime/argrecorder.py

```diff
@@ -304,20 +379,60 @@ def simplify(self, samples):
         self.update_times()
         sample_nodes = self.get_nodes(samples)
         if self.timings is not None:
-            start = timer.process_time()
-        msprime.sort_tables(nodes=self.nodes, edges=self.edges,
-                            sites=self.sites, mutations=self.mutations)
-        #                   migrations=self.migrations)
+            before = timer.process_time()
+        i = self.last_sorted_edge - 1
+        while True:
+            if self.__edges.parent[i] not in sample_nodes:
+                break
+            i -= 1
+        if  i < self.edges.num_rows:
```

Comment:
I've tried to modify the last_sorted_edge to reflect my realization about overlapping gens (here https://github.com/ashander/ftprime/pull/55#discussion_r152161189 ) but does not work

Reply:
oh, right!  doh; should have consulted our former selves, etc.

You cannot do chunkwise sort with overlapping generations.  You'll have to sort the entire table.  Time-reversing before appending and sorting might help because the edges will be close to sorted; but also it may not.

Reply:
right on. I do wish our former selves were more collaborative